frame/balances/src/lib.rs

```diff
@@ -722,6 +725,12 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 	}
 }
 
+impl<T: Trait<I>, I: Instance> OnReapAccount<T::AccountId> for Module<T, I> {
+	fn on_reap_account(who: &T::AccountId) {
+		Locks::<T, I>::remove(who);
```

Comment:
```suggestion
		Locks::<T, I>::remove(who);
		Account::<T, I>::remove(who);
```

I think technically we need to remove `Account` too.